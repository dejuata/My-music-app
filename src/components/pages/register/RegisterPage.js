import React from 'react'
import { Logo } from '../../atoms/Logo/Logo'
import { RegisterForm } from '../../organisms/RegisterForm/RegisterForm'

export const RegisterPage = () => {
    return (
        <div className='auth-form__container'>
            <Logo />

            <RegisterForm />

        </div>
    )
}
