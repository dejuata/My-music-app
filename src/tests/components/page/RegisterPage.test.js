import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Provider } from "react-redux";

import { MemoryRouter } from 'react-router-dom';
import { RegisterPage } from '../../../components/pages/register/RegisterPage';

const middlewares = [ thunk ]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);

const wrapper = mount(
    <Provider store={ store }>
        <MemoryRouter>
            <RegisterPage />
        </MemoryRouter>
    </Provider>
)

describe('test LoginPage', () => {

    beforeEach( () => {
        store = mockStore(initState)
    })

    test('Should be render RegisterPage', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be render <Logo />', () => {

        expect( wrapper.find('Logo').exists() ).toBe( true )

    })

    test('Should be render <RegisterForm />', () => {

        expect( wrapper.find('RegisterForm').exists() ).toBe( true )

    })

})