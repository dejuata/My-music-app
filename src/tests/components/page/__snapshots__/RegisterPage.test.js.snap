// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoginPage Should be render RegisterPage 1`] = `
<div
  className="auth-form__container"
>
  <div
    className="a-logo-container"
  >
    <div
      className="a-logo-container__image"
    >
      <img
        alt="logo"
        src="/assets/logo.svg"
      />
    </div>
    <p
      className="a-logo-container__title"
    >
      MY MUSIC APP
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="alert alert-error"
    />
    <div
      className="form-control mb-2"
    >
      <label
        className="label"
      >
        Name
      </label>
      <input
        autoComplete="off"
        className="input"
        name="name"
        onChange={[Function]}
        placeholder="Your name"
        type="text"
        value="Juan David"
      />
    </div>
    <div
      className="form-control mb-2"
    >
      <label
        className="label"
      >
        Email address
      </label>
      <input
        autoComplete="off"
        className="input"
        name="email"
        onChange={[Function]}
        placeholder="example@email.com"
        type="email"
        value="demo1@gmail.com"
      />
    </div>
    <div
      className="form-control mb-2"
    >
      <label
        className="label"
      >
        Password
      </label>
      <input
        autoComplete="off"
        className="input"
        name="password"
        onChange={[Function]}
        placeholder="********"
        type="password"
        value="123456"
      />
    </div>
    <div
      className="form-control mb-3"
    >
      <label
        className="label"
      >
        Confirm Password
      </label>
      <input
        autoComplete="off"
        className="input"
        name="password2"
        onChange={[Function]}
        placeholder="********"
        type="password"
        value="123456"
      />
    </div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Sign Up
    </button>
    <a
      className="auth__link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered ?
    </a>
  </form>
</div>
`;
